### PR TITLE
commands: prepare ctdb commands for using cluster meta objects

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Preparatory work for being able to store this cluster metadata in rados.

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
cc @avanthakkar

**Review by synarete — DISMISSED** [review]
lgtm

File: sambacc/commands/ctdb.py [hunk 1]
```diff
@@ -86,6 +88,13 @@ def _ctdb_general_node_args(parser: argparse.ArgumentParser) -> None:
         "--persistent-path",
         help="Path to a persistent path for storing nodes file",
     )
+    parser.add_argument(
+        "--metadata-state",
```
Comment (anoopcs9) [inline sambacc/commands/ctdb.py:92]: I think `--metadata-state` doesn't fully match the intended type of value which basically points to an endpoint as source of truth. Isn't it better to use `--metadata-source` or similar?

File: sambacc/commands/ctdb.py [hunk 2]
```diff
@@ -156,6 +173,49 @@ def identity(self):
...
+    @property
+    def cluster_meta_uri(self) -> str:
+        """Return a cluster meta uri value."""
+        values = (
+            # cli takes highest precedence
+            self._ctx.cli.metadata_state,
+            # _nodes_json should only be set if user set it using the old key
+            self._nodes_json,
+            # default or customized value on current key
+            self._cluster_meta_uri,
```
Comment (anoopcs9) [inline sambacc/commands/ctdb.py:185]: May be reverse the order here to prefer new method?

File: sambacc/commands/ctdb.py (hunk 1 above)
Comment (phlogistonjohn) [reply sambacc/commands/ctdb.py:92]: Sure. Will update.

File: sambacc/ctdb.py [hunk 3]
```diff
@@ -338,9 +338,23 @@ def manage_nodes(
     pause_func: typing.Callable,
```
Comment (anoopcs9) [inline sambacc/ctdb.py:338]: `manage_nodes()` now exists for testing purpose only, right?

**Review by anoopcs9 — APPROVED** [review]
See below for a small remark.

Otherwise lgtm, thanks.

**Review by synarete — APPROVED** [review]
lgtm

File: sambacc/ctdb.py (hunk 3 above)
Comment (phlogistonjohn) [reply sambacc/ctdb.py:338]: Yeah, there are now a bunch of functions that are consumed only by the test suite. It's a TODO item to circle back and update the tests and eventually remove the functions.

**Pull request merged**